test/common/testlib.py

```diff
@@ -1129,6 +1149,20 @@ def skipPackage(*args):
     return lambda func: func
 
 
+def nondestructive(testEntity):
+    """Tests decorated as nondestructive will run against the same VM.
+    Can be used on test classes and individual test methods.
+    """
+    def predicate(member):
+        return inspect.isfunction(member) and member.__name__.startswith("test")
+    if inspect.isclass(testEntity):
+        for test_function in inspect.getmembers(testEntity, predicate):
+            test_function[1]._testlib__non_destructive = True
+    else:
```

Comment:
I'd rename `predicate` to `is_test_function` (or so) and apply it here as an `elif:`, throwing an assert in the (now hopefully unreached) `else:` case to be added below